src/monitor-function-calls/agent.js

```diff
@@ -0,0 +1,154 @@
+const {
+  Finding, FindingSeverity, FindingType, ethers,
+} = require('forta-agent');
+
+const {
+  parseExpression,
+  checkLogAgainstExpression,
+  getAbi,
+  extractFunctionArgs,
+} = require('../utils');
+
+// helper function to create alerts
+function createAlert(
+  functionName,
+  contractName,
+  contractAddress,
+  functionType,
+  functionSeverity,
+  args,
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation,
+  expression,
+) {
+  const functionArgs = extractFunctionArgs(args);
+
+  const finding = {
+    name: `${protocolName} Function Call`,
+    description: `The ${functionName} function was invoked in the ${contractName} contract`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-FUNCTION-CALL`,
+    type: FindingType[functionType],
+    severity: FindingSeverity[functionSeverity],
+    protocol: protocolName,
+    metadata: {
+      contractName,
+      contractAddress,
+      functionName,
+      ...functionArgs,
+    },
+  };
+
+  if (expression !== undefined) {
+    finding.description += `, condition met: ${expression}`;
+  }
+
+  return Finding.fromObject(finding);
+}
+
+
+const initialize = async (config) => {
+  let agentState = {...config};
+
+  agentState.contractInfo = config.contracts;
+  agentState.developerAbbreviation = config.developerAbbreviation;
+  agentState.protocolName = config.protocolName;
+  agentState.protocolAbbreviation = config.protocolAbbreviation;
+
+  const contractNames = Object.keys(agentState.contractInfo);
+  agentState.contracts = contractNames.map((name) => {
```

Comment:
Just a simple code cleanup from the original implementation.  The variable `contractNames` isn't used outside of these two lines, as far as I can tell.  It seems like it probably doesn't need to be there.
```suggestion
  agentState.contracts = Object.keys(agentState.contractInfo).map((name) => {
```